src/Plugin/Validation/Constraint/ValidOgRoleConstraint.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Drupal\og\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Ensures that a valid OG role for the group entity type/bundle is selected.
+ *
+ * @Constraint(
+ *   id = "ValidOgRole",
+ *   label = @Translation("OG Role valid reference", context = "Validation")
+ * )
+ */
+class ValidOgRoleConstraint extends Constraint {
+
+  /**
+   * Not a valid role message.
+   *
+   * @var string
+   */
+  public $NotValidRole = 'Invalid role selected.';
```

Comment:
Class property $NotValidRole should use lowerCamel naming without underscores:
```suggestion
  public $notValidRole = 'Invalid role selected.';
```